lib/features/composer/domain/usecases/save_composer_cache_with_identity_on_web_interactor.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:core/presentation/state/failure.dart';
+import 'package:core/presentation/state/success.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/user_name.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/save_composer_cache_on_web_interactor.dart';
+import 'package:tmail_ui_user/features/composer/presentation/model/create_email_request.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/repository/composer_cache_repository.dart';
+import 'package:tmail_ui_user/features/mailbox_dashboard/domain/state/save_composer_cache_with_identity_state.dart';
+
+class SaveComposerCacheWithIdentityOnWebInteractor {
+  SaveComposerCacheWithIdentityOnWebInteractor(
+    this._saveComposerCacheOnWebInteractor,
+    this._composerCacheRepository);
+
+  final SaveComposerCacheOnWebInteractor _saveComposerCacheOnWebInteractor;
+  final ComposerCacheRepository _composerCacheRepository;
+
+  Future<Either<Failure, Success>> execute(
+    CreateEmailRequest createEmailRequest,
+    AccountId accountId,
+    UserName userName
+  ) async {
+    final cacheWithoutIdentityResult = await _saveComposerCacheOnWebInteractor
+      .execute(createEmailRequest);
+    
+    return cacheWithoutIdentityResult.fold(
+      (failure) => Left(
+        SaveComposerCacheWithIdentityFailure(
+          failure is FeatureFailure
+          ? failure.exception
+          : null)),
+      (success) {
+        _composerCacheRepository.markComposerCacheIdentityOnWeb(accountId, userName);
```

Comment:
I think about some risks in separating steps in cache of composer and identity. How about to store it in the same place?

Reply:
That would require a modification of `SaveComposerCacheOnWebInteractor`. Will proceed if that's ok

Reply:
no problem for me